人工智能/人工智能数学基础/人智数学基础实验/2026春实验一/实验一代码/无外点ransac拟合圆.py

```
@@ -0,0 +1,141 @@
+    for it in range(max_iter):#max_iter轮
+        # 随机采样三点
+        idx = random.sample(range(n), 3)
+        sample_points = points[idx]  # 保存这三个点
+        p1, p2, p3 = points[idx[0]], points[idx[1]], points[idx[2]]
+        # 拟合圆
+        circle = fit_circle_three_points(p1, p2, p3)
+        if circle is None:
+            continue
+        
+        # 统计内点
+        distances = np.array([distance_to_circle(p, circle) for p in points])#求出points数组中每个p到圆的距离distances，distances也是一个数组
```

Ruff (0.15.17)

[warning] Comment contains ambiguous `，` (FULLWIDTH COMMA). Did you mean `,` (COMMA)?

(RUF003)